src/core/reducers/addons.js

```diff
@@ -330,6 +356,7 @@ export default function addonsReducer(
 
         if (addon.slug) {
           bySlug[addon.slug] = addon.id;
+          loadingBySlug[addon.slug] = false;
```

Comment:
I realized that doing assignments like this is bad for the same reason as using the [delete statement](https://github.com/mozilla/addons-frontend/issues/6102) -- it breaks the Redux devtool. It would probably be annoying to rewrite this whole switch case so maybe file a follow-up?

Reply:
I might just do it now. Let me see how much time I think it will take.

Reply:
Actually, I'll file a follow-up. It's going to take some work to address this.

Reply:
I opened #6147.

Reply:
How is it breaking the Redux devtools? `loadingBySlug` is a new object created L349.

Reply:
I took a closer look at this specific case and it's safe (I think) but this is a very dangerous pattern so we should avoid it. Here is one small change which would mutate the previous state -- it would be easy to introduce this by accident.

```js
const prevState = {
  byID: {
    '123': { name: 'first' },
  },
};

const nextState = {
  byID: {
    ...prevState.byID,
  },
};
  
nextState.byID['123'].name = 'second';
  
console.log('previous:', prevState.byID[123].name);
console.log('next:', nextState.byID[123].name);
```

Get the [babel translation here](https://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=MYewdgzgLgBADgJwKYDcDKUCGUkwLwwDeAUDDAEYCeAkgCIBcRpZMA5AIwBMAzK44TDCYAtkkasAZgEsE0VjAC-AGmbLiCgNzFioSLDBIAHlAzZcBEmSp1-zMgDpHiVKZz3rtFWTWbmxA8auSO40tADaHDysALr2QqL4bBBIugAmrFpkOuAQIAA2wXkgAOYAFKzOKFIgAK4Q9KxK8MjoWG4eYVzcsfFIAJRaurkF9kVlrAFQDU2TQSF0nTw9Iv0aQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=6.26.0&envVersion=)

When you run this, you can see that the previous state was mutated:

<img width="397" alt="screenshot 2018-09-06 09 55 59" src="https://user-images.githubusercontent.com/55398/45165759-1559af80-b1bb-11e8-85cd-cc30694d3170.png">

This is because objects are passed by "copy of a reference," similar to Python.